core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/SamplesReceivedWorkflow.java

```diff
@@ -0,0 +1,536 @@
+    workflowExecutor.save(sample);
+  }
+  
+  private static class ConcentrationStep implements WorkflowStep {
+
+    private PositiveDoubleProgressStep concStep;
+
+    private final int aliquotIndex;
+
+    ConcentrationStep(int aliquotIndex) {
+      this.aliquotIndex = aliquotIndex;
+    }
+
+    @Override
+    public WorkflowStepPrompt getPrompt() {
+      return new WorkflowStepPrompt(Sets.newHashSet(InputType.POSITIVE_DOUBLE), String.format("Enter the Concentration for Aliquot #%d"
+          + " (in ng/µl)", aliquotIndex + 1));
+    }
+
+    @Override
+    public ProgressStep getProgressStep() {
+      return concStep;
+    }
+
+    @Override
+    public void cancelInput() {
+      concStep = null;
+    }
+
+    @Override
+    public String getLogMessage() {
+      return String.format("Set Aliquot #%d to have a Concentration of %fng/µl", aliquotIndex + 1, getConcentration());
+    }
+
+    @Override
+    public void processInput(PositiveDoubleProgressStep step) {
+      concStep = step;
+    }
+
+    public double getConcentration() {
+      return concStep.getInput();
+    }
+
+  }
+
+  private static class VolumeStep implements WorkflowStep {
+
+    private PositiveDoubleProgressStep volumeStep;
+
+    private final int aliquotIndex;
+
+    VolumeStep(int aliquotIndex) {
+      this.aliquotIndex = aliquotIndex;
+    }
+
+    @Override
+    public WorkflowStepPrompt getPrompt() {
+      return new WorkflowStepPrompt(Sets.newHashSet(InputType.POSITIVE_DOUBLE), String.format("Enter the Volume for Aliquot #%d (in µl)",
+          aliquotIndex + 1));
+    }
+
+    @Override
+    public ProgressStep getProgressStep() {
+      return volumeStep;
+    }
+
+    @Override
+    public void cancelInput() {
+      volumeStep = null;
+    }
+
+    @Override
+    public String getLogMessage() {
+      return String.format("Set Aliquot #%d to have a Volume of %fµl", aliquotIndex + 1, getVolume());
+    }
+
+    @Override
+    public void processInput(PositiveDoubleProgressStep step) {
+      volumeStep = step;
+    }
+
+    public double getVolume() {
+      return volumeStep.getInput();
+    }
+
+  }
+
+  private static class BoxPositionStep implements WorkflowStep {
+
+    private BoxPositionProgressStep boxPositionStep;
+
+    private final int aliquotIndex;
+    private final Box box;
+    private String firstFreePosition;
+    private final Set<String> occupiedLocations;
+
+    private String error;
+
+    BoxPositionStep(int aliquotIndex, Box box, Set<String> occupiedLocations) {
+      this.aliquotIndex = aliquotIndex;
+      this.box = box;
+      this.occupiedLocations = occupiedLocations;
+    }
+
+    @Override
+    public WorkflowStepPrompt getPrompt() {
+      firstFreePosition = getFirstFreePosition();
+      return new WorkflowStepPrompt(Sets.newHashSet(InputType.BOX_POSITION, InputType.SKIP),
+          String.format("Enter a Box Position to place Aliquot #%d.\nSkipping will default to position %s", aliquotIndex + 1,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') %n should be used in place of \n to produce the platform-specific line separator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)